src/main/java/org/jenkinsci/plugins/authorizeproject/AuthorizeProjectProperty.java

```diff
@@ -146,43 +139,13 @@ public static void setStrategyCritical() {
     }
 
     /**
-     * If we are being deserialized outside of loading the initial jobs (or reloading) then we need to cross check
-     * the strategy permissions to defend against somebody trying to push a configuration relating to a user other
-     * than themselves.
-     *
-     * @return {@code this}
-     * @throws ObjectStreamException if the object cannot be deserialized.
-     * @see #setStrategyCritical()
-     */
-    private Object readResolve() throws ObjectStreamException {

```

Comment:
Move to `AuthorizeProjectStrategy`, as `#setStrategyCritical` makes exceptions from `AuthorizeProjectStrategy#readResolve` critical but doesn't make exceptions from `AuthorizeProjectProperty#readResolve` critical.